### PR TITLE
Martijn screenbuffer

### PR DESCRIPTION
**Pull request by Mfrielink**

### CODE REVIEW

File: src/engine/SDLFacade.cpp [hunk 1]
```diff
@@ -100,8 +104,8 @@ namespace Engine {
     /// \return This function returns True if the _window was successfully initialized, ohterwise it returns False
     bool SDLFacade::_init_window()
     {
-        this->_window = SDL_CreateWindow(Config::GAME_WINDOW_TITLE.c_str(), SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 500,
-                                   500, SDL_WINDOW_SHOWN);
+        this->_window = SDL_CreateWindow("Vidya game", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, this->_width,
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Why is this changed back to a magic constant? The config thing is intended :p

File: src/engine/SDLFacade.cpp [hunk 2]
```diff
@@ -111,6 +115,21 @@ namespace Engine {
             return false;
         }
     }
+    /// \brief Initialiser for the _screen_buffer (SDL_Texture)
+    ///
+    /// \return This function returns True if the _screen_buffer was successfully initialized, ohterwise it returns False
+    bool SDLFacade::_init_screen_buffer()
+    {
+        this->_screen_buffer = SDL_CreateTexture(this->_renderer, SDL_PIXELFORMAT_ARGB8888, SDL_TEXTUREACCESS_STREAMING,
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Are you allowed to pick this PIXELFORMAT yourself? shouldn't you stick to what the window / renderer uses?

File: src/engine/SDLFacade.cpp [hunk 3]
```diff
@@ -111,6 +115,21 @@ namespace Engine {
...
+                                                 this->_width, this->_height);
+
+        if(this->_screen_buffer != nullptr){
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: spaces around the `()`.
Like this `if (foo) {`

File: src/engine/SDLFacade.cpp [hunk 4]
```diff
@@ -111,6 +115,21 @@ namespace Engine {
...
+            return true;
+        }else{
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: same here

File: src/engine/SDLFacade.cpp [hunk 5]
```diff
@@ -132,6 +151,38 @@ namespace Engine {
         SDL_RenderDrawLine(this->_renderer, line_start.x, line_start.y, line_end.x, line_end.y);
     }
 
+    /// \brief Draws a single pixel
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: This doesn't draw a pixel, it changes the pixel in the buffer. the pixel is not drawn until it's `RenderCopy`'ed

File: src/engine/SDLFacade.cpp [hunk 6]
```diff
@@ -132,6 +151,38 @@ namespace Engine {
...
+
+        int pixel_location = position.x + (position.y * this->_width);
+        buffer_pixels[pixel_location] = pixel;
+
+        SDL_UnlockTexture(_screen_buffer);
+    }
+
+    /// \brief Update of _screen_buffer
+    ///
+    /// Copies the drawn _screen_buffer (SDL_Texture) to the renderer
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: The `_screen_buffer` is not yet drawn here.

File: src/engine/SDLFacade.cpp [hunk 7]
```diff
@@ -132,6 +151,38 @@ namespace Engine {
...
+    void SDLFacade::update_screen_buffer()
+    {
+        SDL_RenderCopy(this->_renderer, _screen_buffer, NULL, NULL);
+
+        SDL_DestroyTexture(this->_screen_buffer);
+
+        this->_screen_buffer = SDL_CreateTexture(this->_renderer, SDL_PIXELFORMAT_ARGB8888, SDL_TEXTUREACCESS_STREAMING,
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Is this the most efficient way to empty the screen buffer?

File: src/main.cpp [hunk 8]
```diff
@@ -6,6 +6,5 @@ int main()
 {
     Game game;
     game.run();
-    return 0;
```
Comment (ThomasvLingen) [inline src/main.cpp]: This should definitely stay there.
Now we have an undefined return value of our program!

File: src/engine/SDLFacade.cpp (hunk 7 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Fair enough

File: src/engine/SDLFacade.cpp (hunk 2 above)
Comment (Mfrielink) [inline src/engine/SDLFacade.cpp]: Yes, this would be better. However, I can't find a clear solution to the problem. We'll have to ask, perhaps in the next showcase.

File: src/engine/SDLFacade.cpp (hunk 1 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: This is still not fixed

File: src/engine/SDLFacade.cpp (hunk 2 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Hmm, annotate it in a Github issue

File: src/engine/SDLFacade.cpp (hunk 3 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: done

File: src/engine/SDLFacade.cpp (hunk 4 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: not fixed yet

File: src/engine/SDLFacade.cpp (hunk 5 above)
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: done
